backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/gson/spi/JsonLogHelper.java

```diff
@@ -19,7 +19,7 @@
  *
  * @author Yoann Rodiere
  */
-public class JsonLogHelper {
+public final class JsonLogHelper {
```

Comment:
Same question here, what's the purpose of exposing a helper class you cannot extend and that does not implement an interface (so you can't provide an alternative implementation)?

Reply:
The helper is provided for use by the Elasticsearch client (since it's... a helper). It's not supposed to be implemented by anyone else, and if people don't want to use it, they can ignore it.

The helper is only here to help logging the requests/responses, though. I could improve the situation: I could only pass a Gson object to the client (not a GsonProvider), and have Hibernate Search wrap all clients with a wrapper that implements the logging. Something like this:

```
public class LoggingElasticsearchClientWrapper implements ElasticsearchImplementor {
    CompletableFuture<ElasticsearchResponse> submit(ElasticsearchRequest request) {
        int startTime = ...;
        CompletableFuture<ElasticsearchResponse> result = delegate.submit( request );
        if ( requestLog.isDebugEnabled() ) {
            result.thenAccept( response -> log( request, startTime, response ) );
        }
        return result;
    }

    // ... delegate the other methods...

	private void log(ElasticsearchRequest request, long start, ElasticsearchResponse response) {
            // Log according to backend settings (passed to the constructor)
	}
}
```

Then the SPI would only be tainted with `Gson`, not with our own classes (`GsonProvider` and `JsonHelper`).